agent/taskresource/volume/dockervolume.go

```diff
@@ -38,19 +34,36 @@ type VolumeResource struct {
 	lock sync.RWMutex
 }
 
+// DockerVolumeConfig represents docker volume configuration
```

Comment:
Consider:

a) linking to docs explaining what these are
b) commenting on each field

Reply:
Added link to the docker API for volume, added comments for 2 fields we introduced (scope and autoprovision).